### PR TITLE
Fix keyboard interactivity for Menu

### PR DESCRIPTION
**Pull request by taysea**
#### What does this PR do?
Changes functionality of Menu with keyboard interactions by removing restrictFocus from DropButton. Allows users to use up/down arrow keys to open/navigate through Menu component. With restrictFocus set to true, it puts the Drop in control of focus— meaning that "tab" must be used to advance to the next tabbable element within the focused component. In this case, that made tab the only way users could advance between menu items. By removing "restrictFocus", the menu doesn't become a modal that controls the focus, and because of this the arrow keys can be used to properly navigate the menu items.

#### Where should the reviewer start?
js/components/DropButton.js

#### What testing has been done on this PR?
Testing was done in storybook to ensure functionality of other components such as Select or DropButton were not affected.

#### How should this be manually tested?
Testing can be done using Storybook components by interacting with keyboard navigation in Menu, Select, and DropButton.

#### Any background context you want to provide?
Based on HTML behaviors, up and down arrows should allow users to open and navigate within a menu. However, the Menu component was not allowing this and was incorrectly requiring users to use tab within a menu.

#### What are the relevant issues?
Issue #3225 

#### Do the grommet docs need to be updated?
No.

#### Is this change backwards compatible or is it a breaking change?
If users were previously using tab to navigate within a menu component, they will now need to use arrow keys. However, upon further investigation it appears that tab is the correct behavior. Therefore, this PR maybe should be disregarded as the existing grommet behavior seems to be correct.

### CODE REVIEW

File: src/js/components/Select/__tests__/__snapshots__/Select-test.js.snap [hunk 1]
```diff
@@ -1968,7 +1968,267 @@ exports[`Select disabled 2`] = `
 `;
 
 exports[`Select empty results search 1`] = `
+<body>
```
Comment (oorestisime) [inline src/js/components/Select/__tests__/__snapshots__/Select-test.js.snap]: mmmm why we have all this changed with the restrictFocus ?? something seems off to me here

**Comment by oorestisime** [comment]
hey thanks for the contribution, left a small comment :)
Keep in mind that this branch will also need to be rebased to current master

File: src/js/components/Select/__tests__/__snapshots__/Select-test.js.snap (hunk 1 above)
Comment (taysea) [reply src/js/components/Select/__tests__/__snapshots__/Select-test.js.snap]: @oorestisime Just to clarify, are you asking why this test snapshot changed so much just with the removal of "restrictFocus" from DropButton?

Comment (ShimiSun) [reply src/js/components/Select/__tests__/__snapshots__/Select-test.js.snap]: @taysea usually when we have snapshot changes it means the fix might not be backward compatible.

Comment (oorestisime) [reply src/js/components/Select/__tests__/__snapshots__/Select-test.js.snap]: yeah plus the diff seems unrelated since it is in the select.

Comment (taysea) [reply src/js/components/Select/__tests__/__snapshots__/Select-test.js.snap]: Okay, let me dig a little deeper to check out what is going on. Thanks for the feedback!

Comment (taysea) [reply src/js/components/Select/__tests__/__snapshots__/Select-test.js.snap]: After further reading, it seems that in navigation menus, the typical behavior is that "tab" is used to navigate between elements. This differs from the behavior of Select, which allows arrows to navigate between elements. If we want to remain with convention on how keyboard interactions should behave in Menus, then I think this entire PR should be disregarded and the existing grommet behavior should stay. For more, see this article: https://hiddedevries.nl/en/blog/2017-01-29-using-javascript-to-trap-focus-in-an-element

Comment (oorestisime) [reply src/js/components/Select/__tests__/__snapshots__/Select-test.js.snap]: @ericsoderberghp we need your input here :)

Comment (ShimiSun) [reply src/js/components/Select/__tests__/__snapshots__/Select-test.js.snap]: @taysea does it mean that your comment about backward compatibility needs to be refined and it is not really the correct behavior:
>If users were previously using tab to navigate within a menu component, they will now need to use arrow keys. However, arrow keys are the correct behavior.

Comment (taysea) [reply src/js/components/Select/__tests__/__snapshots__/Select-test.js.snap]: @ShimiSun You're right. Thank you. I've updated now.

Comment (oorestisime) [reply src/js/components/Select/__tests__/__snapshots__/Select-test.js.snap]: I am confused now. should we drop this PR?

Comment (ShimiSun) [reply src/js/components/Select/__tests__/__snapshots__/Select-test.js.snap]: I think we should drop it, but I want @ericsoderberghp to be aware of it

Comment (taysea) [reply src/js/components/Select/__tests__/__snapshots__/Select-test.js.snap]: I agree. @ShimiSun

**Comment by ShimiSun** [comment]
@taysea thank you for doing the investigation, although this PR won't be merged the investigation itself was a great contribution. 💟

**Pull request closed without merging**